src/ports/postgres/modules/mxgboost/madlib_xgboost.py_in

```diff
@@ -0,0 +1,549 @@
+
+    #run weights
+    sample_representation = y_train.value_counts()
+    total_samples = sum(sample_representation)
+    sample_weights = None
+    if not class_weights:
+        sample_weights = map(
+                lambda s: total_samples*1.0/sample_representation[s]
+                                /
+                sum([total_samples*1.0/sample_representation[c] for c in sample_representation.keys()])
+                ,
+                y_train
+            )
+    else:
+        #User-supplied class-weights
+        class_weights_dict = ast.literal_eval(re.sub("[\\t]","",class_weights).strip())
+        sample_weights = map(lambda s: class_weights_dict[s], y_train)
+
+    #Train gradient boosted trees
+    p_list = [p.split('=') for p in ast.literal_eval(re.sub("[\\t]","",params).strip())]
+    params_dict = dict([(k, ast.literal_eval(v.strip())) for k,v in p_list])
+    gbm = xgb.XGBClassifier(**params_dict)
+
+    #Fit model
+    gbm.fit(
+        X_train,
+        y_train,
+        eval_metric = 'auc',
+        sample_weight = sample_weights
+    )
+    #Compute and return model metrics score
+    y_pred_train = gbm.predict(X_train)
+    y_pred_test = gbm.predict(X_test)
+    cmat_train = confusion_matrix(y_train, y_pred_train)
+    cmat_test = confusion_matrix(y_test, y_pred_test)
+    scores = numpy.array(precision_recall_fscore_support(y_test, y_pred_test)).transpose()
+
+    metric_labels = ['precision', 'recall', 'fscore', 'support']
+    model_metrics = print_prec_rec_fscore_support(scores, metric_labels, class_label, gbm.classes_)
+
+    #Calculate feature importance scores
+    importance = gbm._Booster.get_fscore()
+    if len(importance) == 0:
+        plpy.error("No importance found for any feature")
+    fnames_importances = sorted(
+        [(features[int(k.replace('f',''))], importance[k]) for k in importance],
+        key=itemgetter(1),
+        reverse=True
+    )
+    fnames, f_importance_scores = zip(*fnames_importances)
+    important_features = pd.DataFrame(fnames_importances)
+
+    test_ids = [int(x) for x in test_ids]
+    return (features, pickle.dumps(gbm), params, fnames, f_importance_scores,
+        model_metrics.iloc[:,1].values.tolist(), model_metrics.iloc[:,2].values.tolist(),
+        model_metrics.iloc[:,3].values.tolist(),model_metrics.iloc[:,4].values.tolist(),
+        test_ids)
+
+def xgboost_grid_search(schema_madlib, source_table, id_column, class_label,
+                        list_of_features, list_of_features_to_exclude,
+                        params_str, grid_search_results_tbl, class_weights,
+                        train_set_size, train_set_split_var):
+
+    _assert(table_exists(source_table), "Input table missing")
+    _assert(not table_exists(grid_search_results_tbl), "Output table exists")
+    grid_search_results_tbl_summary = add_postfix(grid_search_results_tbl, '_summary')
+    _assert(not table_exists(grid_search_results_tbl_summary), "Output table exists")
```

Comment:
We should also display the name of the output table that exists.

Reply:
I'll update all of these messages to bring them in line with the rest of the modules.